paperoni/sources/scrapers/openreview.py

```diff
@@ -66,87 +149,29 @@ def query(
-                            _links.append(
-                                Link(type="email", link=authors_email)
-                            )
-                        authors.append(
-                            PaperAuthor(
-                                affiliations=[],
-                                author=Author(
-                                    name=name,
-                                    affiliations=[],
-                                    aliases=[],
-                                    links=_links,
-                                    roles=[],
-                                ),
-                            )
-                        )
-                    _links = [Link(type="openreview", link=note.id)]
-                    if "code" in note.content:
-                        Link(type="git", link=note.content["code"])
-                    yield Paper(
-                        title=note.content["title"],
-                        abstract=note.content.get("abstract", ""),
-                        citation_count=0,
-                        authors=authors,
-                        releases=[
-                            Release(
-                                venue=Venue(
-                                    type=venue_type,
-                                    name=note.content["venue"],
-                                    links=[],
-                                ),
-                                date=str(
-                                    datetime.fromtimestamp(note.tcdate / 1000)
-                                ),
-                                date_precision=DatePrecision.day,
-                            )
-                        ],
-                        topics=[
-                            Topic(name=kw)
-                            for kw in note.content.get("keywords", [])
-                        ],
-                        links=_links,
-                        scrapers=["orev"],
-                    )
-                next_offset += len(notes)
-                if not notes:
-                    break
-            total += next_offset
+            for paper in OpenReviewScraper._query(client, params, total, limit):
+                total += 1
+                yield paper
 
     @tooled
     def acquire(self, queries):
-        pass
+        todo = {}
+
+        for auq in queries:
+            for link in auq.author.links:
+                if link.type == "openreview":
+                    todo[link.link] = auq
+
+        client = openreview.Client(baseurl="https://api.openreview.net")
+
+        for author_id, auq in todo.items():
+            print(f"Fetch papers for {auq.author.name} (email={author_id})")
+            params = {
+                "content": {"authorids": [author_id]},
+                "mintcdate": int(auq.start_date.timestamp() * 1000),
+            }
+            for paper in OpenReviewScraper._query(client, params):
+                yield paper
```

Comment:
Not exactly sure how this goes. Let me know if there's something going bad here, I need to talk a bit longer with you about this to really understand the flow.

Let me know also if you would like to skip papers which are after `auq.end`

This will most likely skip quite a bit of papers since `authorids` is not always used. Should we implement something to palliate (something based on names and cluster of related names like Mila's professors?) or is this ok for now?

Reply:
> Let me know also if you would like to skip papers which are after auq.end

If there's no way to query using a maxtcdate, we can ignore auq.end and filter out the papers in the database post facto.

> This will most likely skip quite a bit of papers since authorids is not always used.

We already have paper data prior to 2022, so as long as authorids are available for all papers from 2022, it shouldn't matter.